src/python/grpcio/grpc/_cython/_cygrpc/records.pyx.pxi

```diff
@@ -235,18 +235,13 @@ cdef class ByteBuffer:
     if data is None:
       self.c_byte_buffer = NULL
       return
-    if isinstance(data, bytes):
-      pass
-    elif isinstance(data, basestring):
-      data = data.encode()
-    elif isinstance(data, ByteBuffer):
+    if isinstance(data, ByteBuffer):
```

Comment:
Hmm, the lack of uniformity of strings and bytes and how that maps into `str_to_bytes` thickens...

I wasn't sure what the most aesthetically pleasing answer was back when I first wrote the code (more functions? less functions? best-effort approach with one function?) and I'm still not sure now.